v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/ISchemaMapper.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.schema;
+
+import com.google.cloud.teleport.v2.spanner.type.Type;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public interface ISchemaMapper {
```

Comment:
Is this only for MySQL? 
Different source databases will have different modeling concepts. For example, PG will have 

database -> schema -> table -> column

How will `ISchemaMapper` work for that case?

Reply:
We currently do have a schema field but we just populate it with the db name instead of the namespace. The namespace is handled by adding modifying the table name to `namespace.tablename`.

 The reader plans on supporting namespaces first-hand, but the transformer will have to concat it for postgres handling as long as the session file is modified the way it is currently.

It might make sense to add namespace as an input to the interface and pass null for now.

Reply:
Done